File: sast-java/src/main/java/com/sast/astbenchmark/case_language_maturity/accuracy/object_field_sensitive/object_sensitive/map/Map_obj_sensitive_001_T.java
package com.sast.astbenchmark.case_language_maturity.accuracy.object_field_sensitive.object_sensitive.map;

import com.sast.astbenchmark.common.utils.SinkUtil;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

/**
 * Introduction 准确度->对象敏感与域敏感分析->区分不同的类对象、结构体/联合体和字典/列表/数组->容器->Map
 * Level 2
 * Date 2024-06-28
 */
// evaluation information start
// real case = true
// evaluation item = 准确度->对象敏感与域敏感分析->区分不同的类对象、结构体/联合体和字典/列表/数组
// bind_url = accuracy/object_field_sensitive/object_sensitive/map/Map_obj_sensitive_001_T
// evaluation information end
@RestController()
@RequestMapping("accuracy/object_field_sensitive/object_sensitive/map")
public class Map_obj_sensitive_001_T {
    @PostMapping("Map_obj_sensitive_001_T")
    public Map<String, Object> testcase(@RequestParam String cmd) {
        Map<String, Object> modelMap = new HashMap<>();
        try {
            Map<String, String> m = new HashMap<>();
            m.put("key1", cmd);
            SinkUtil.sink(m);
            modelMap.put("status", "success");
        } catch (Exception e) {
            modelMap.put("status", "error");
        }
        return modelMap;
    }
}

File: sast-go/cases/accuracy/object_sensitive/collection/map_obj_sensitive_001_T/map_obj_sensitive_001_T.go

// evaluation information start
// real case = true
// evaluation item = 准确度->对象敏感与域敏感分析->区分字典/列表/数组的不同元素
// scene introduction = map->对象敏感
// level = 2
// bind_url = accuracy/object_sensitive/collection/map_obj_sensitive_001_T/map_obj_sensitive_001_T
// evaluation information end

package main
import (
	"os/exec"
	"fmt"
)

func map_obj_sensitive_001_T(__taint_src string) {
	m := make(map[string]string, 1)
	m2 := make(map[string]string, 1)
	m["key1"] = __taint_src
	m2["key1"] = "value"
	__taint_sink(m)
}

func __taint_sink(o interface{}) {
	_ = exec.Command("sh", "-c", fmt.Sprintf("%v", o)).Run()
	}

func main() {
    __taint_src := "taint_src_value"
    map_obj_sensitive_001_T(__taint_src)
}